From: Matthias Klose <doko@debian.org>
Date: Sat, 29 Oct 2022 21:23:31 -0400
Subject: distutils: Don't add standard library dirs to library_dirs and
 runtime_library_dirs.

On amd64, runtime paths pointing to /usr/lib64 aren't recognized by
dpkg-shlibdeps, and the packages containing these libraries aren't added to
${shlibs:Depends}.
Forwarded: no
---
 Lib/distutils/unixccompiler.py | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/Lib/distutils/unixccompiler.py b/Lib/distutils/unixccompiler.py
index d00c489..70b4cb5 100644
--- a/Lib/distutils/unixccompiler.py
+++ b/Lib/distutils/unixccompiler.py
@@ -155,6 +155,17 @@ class UnixCCompiler(CCompiler):
                                         runtime_library_dirs)
         libraries, library_dirs, runtime_library_dirs = fixed_args
 
+        # filter out standard library paths, which are not explicitely needed
+        # for linking
+        system_libdirs = ['/lib', '/lib64', '/usr/lib', '/usr/lib64']
+        multiarch = sysconfig.get_config_var("MULTIARCH")
+        if multiarch:
+            system_libdirs.extend(['/lib/%s' % multiarch, '/usr/lib/%s' % multiarch])
+        library_dirs = [dir for dir in library_dirs
+                        if not dir in system_libdirs]
+        runtime_library_dirs = [dir for dir in runtime_library_dirs
+                                if not dir in system_libdirs]
+
         lib_opts = gen_lib_options(self, library_dirs, runtime_library_dirs,
                                    libraries)
         if not isinstance(output_dir, (str, type(None))):
